### DISCUSSION TITLE
Which approach should I follow?

### DISCUSSION THREAD

**Discussion by rahuldev-2021** [ultralytics/ultralytics#16316] [Q&A]
Using a dataset with ten classes, I trained a Yolov8s-world-v2 model for real-time object detection. My goal is to continuously add additional classes without having to train the model again from scratch (**no forgetting of pre-trained classes**).

**Training the model with only new class**

I've tried changing batch layers, freezing layers, and varying the learning rate, but none of them have produced the results I was hoping for. Could you suggest a method for achieving incremental learning that avoids the need for the model to be completely retrained from scratch?

For training the model with the new class alone, How many instances(data)are required in each - class for every incremental training?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rahuldev-2021, thank you for reaching out to the Ultralytics community 🚀! It sounds like you're delving into the exciting area of incremental learning with your YOLOv8s-world-v2 model.

For comprehensive guidance, please explore our [Docs](https://docs.ultralytics.com), where a wealth of resources like [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples are available.

If this is a 🐛 Bug Report, we kindly request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in debugging the issue.

If your inquiry pertains to custom training ❓, please provide detailed information, including dataset samples and training logs. Also, ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our vibrant Ultralytics community for real-time discussions on [Discord](https://ultralytics.com/discord) 🎧, or partake in extended discussions on [Discourse](https://community.ultralytics.com). Our [Subreddit](https://reddit.com/r/ultralytics) is also a great place to exchange insights with fellow developers.

## Upgrade

Make sure you have the latest version of the `ultralytics` package, including [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Use [**Python>=3.8**](https://www.python.org/) with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 runs in several verified environments with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn):

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: Follow the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>
  
When this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, ensuring smooth functionality of YOLOv8 across multiple platforms.

Please note that this is an automated response. An Ultralytics engineer will attend to your query shortly. 😊

**Comment by glenn-jocher** [comment]
@rahuldev-2021 to achieve incremental learning without forgetting pre-trained classes, consider using techniques like knowledge distillation or fine-tuning with a balanced dataset that includes both old and new classes. The number of instances required can vary, but a balanced representation of each class is generally recommended to maintain performance. For more detailed guidance, please refer to our documentation on model training tips.

**Reply by rahuldev-2021** [reply]
I am currently exploring knowledge distillation and fine-tuning techniques, and I have a specific question regarding the dataset preparation process. Most resources suggest that we should combine both old class and new class data when training a model.

Can I train a pre-trained model using just a dataset with new class data alone, without including the old class data?

Also, can you please share me the detailed documentation regarding these approaches

Thanks for your response!

**Reply by glenn-jocher** [reply]
@rahuldev-2021 training with only new class data can lead to forgetting old classes. It's best to include both old and new class data to maintain performance. For detailed guidance, please refer to our documentation on model training techniques.
